src/components/Workspace/WorkspaceEditor.tsx

```diff
@@ -53,15 +54,35 @@ const WorkSpaceEditor: React.FC = () => {
     (state) => state.setCurrentNetworkId,
   )
 
+  const viewModels: Record<string, NetworkView> = useViewModelStore(
+    (state) => state.viewModels,
+  )
+
   const setNetworkModified: (id: IdType, isModified: boolean) => void =
     useWorkspaceStore((state) => state.setNetworkModified)
 
   useViewModelStore.subscribe(
     (state) => state.viewModels[currentNetworkId],
-    () => {
+    (prev: NetworkView, next: NetworkView) => {
```

Comment:
This function is the main piece of logic that determines whether to show/hide the network modified icon.  TODO make this approach more efficient in the future.